src/db.rs

```diff
@@ -1269,6 +1378,188 @@ mod tests {
+        let values = [b"test_value1", b"test_value2", b"test_value3"];
+
+        for i in 0..3 {
+            kv_store.put(keys[i], values[i]).await.unwrap();
+        }
+
+        // advance clock to t=99 --> still returned
+        clock.ticker.store(99, Ordering::SeqCst);
+        assert_eq!(
+            vec![
+                Some(Bytes::from_static(values[0])),
+                Some(Bytes::from_static(values[1])),
+                Some(Bytes::from_static(values[2]))
+            ],
+            kv_store
+                .multi_get_with_options(
+                    &keys,
+                    &ReadOptions {
+                        durability_filter: Memory
+                    }
+                )
+                .await
+                .unwrap(),
+        );
+
+        // advance clock to t=100 --> no longer returned
+        clock.ticker.store(100, Ordering::SeqCst);
+        assert_eq!(
+            vec![None; 3],
+            kv_store
+                .multi_get_with_options(
+                    &keys,
+                    &ReadOptions {
+                        durability_filter: Memory
+                    }
+                )
+                .await
+                .unwrap(),
+        );
+
+        kv_store.close().await.unwrap();
+    }
+
+    #[tokio::test]
+    async fn test_multi_get_with_flush() {
+        let object_store: Arc<dyn ObjectStore> = Arc::new(InMemory::new());
+        let kv_store = Db::open_with_opts(
+            Path::from("/tmp/test_kv_store"),
+            test_db_options(0, 1024, None),
+            object_store,
+        )
+        .await
+        .unwrap();
+
+        let keys = [b"test_key1", b"test_key2", b"test_key3"];
+        let values = [b"test_value1", b"test_value2", b"test_value3"];
+
+        // Insert keys and values
+        for i in 0..3 {
+            kv_store.put(keys[i], values[i]).await.unwrap();
+        }
+
+        // Verify values are in memory
+        assert_eq!(
+            vec![
+                Some(Bytes::from_static(values[0])),
+                Some(Bytes::from_static(values[1])),
+                Some(Bytes::from_static(values[2]))
+            ],
+            kv_store
+                .multi_get_with_options(
+                    &keys,
+                    &ReadOptions {
+                        durability_filter: Memory
+                    }
+                )
+                .await
+                .unwrap(),
+        );
+
+        // Flush to disk
+        kv_store.flush().await.unwrap();
+
+        // Verify values are still accessible after flush
+        assert_eq!(
+            vec![
+                Some(Bytes::from_static(values[0])),
+                Some(Bytes::from_static(values[1])),
+                Some(Bytes::from_static(values[2]))
+            ],
+            kv_store
+                .multi_get_with_options(
+                    &keys,
+                    &ReadOptions {
+                        durability_filter: Memory
```

Comment:
I think it makes more sense to set `durability_filter: Remote` here. Once flushed the data should be visible remotely.